src/main/tools/linux-sandbox-pid1.cc

```diff
@@ -310,11 +364,21 @@ static void MakeFilesystemMostlyReadOnly() {
 }
 
 static void MountProc() {
-  // Mount a new proc on top of the old one, because the old one still refers to
-  // our parent PID namespace.
-  if (mount("/proc", "/proc", "proc", MS_NODEV | MS_NOEXEC | MS_NOSUID,
-            nullptr) < 0) {
-    DIE("mount");
+  if(opt.hermetic){
+    // Create a bind mount to /proc
+    if (CreateTarget("proc", true) < 0) {
+      DIE("CreateTarget");
+    }
+    if (mount("/proc", "proc", NULL, MS_REC | MS_BIND, NULL) < 0) {
+      DIE("mount");
+    }
```

Comment:
What's the reason that we no longer have to mount a new proc on top of the old one, because the old one still refers to our parent PID namespace, but can just use a bind mount in this case?

Reply:
I could have a look, this code is taken from old linux-sandbox implementation, but maybe its not needed.

Reply:
Thanks! I remember this being tricky, but in the end we found a way that worked. 🤔 A good way to check whether /proc is working correctly is that when you manually run `linux-sandbox` e.g. with a shell and you look inside /proc, you shouldn't see any processes from outside the sandbox.

Reply:
Yes it works